examples/Python/ChatApp/app.py

```diff
@@ -0,0 +1,192 @@
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+Azure OpenAI Chat Application using Azure App Configuration.
+This script demonstrates how to create a chat application that uses Azure App Configuration
+to manage settings and Azure OpenAI to power chat interactions.
+"""
+
+import os
+from typing import List, Dict, Any
+from azure.core.credentials import TokenCredential
+from azure.identity import DefaultAzureCredential, get_bearer_token_provider
+from azure.appconfiguration.provider import load, SettingSelector, WatchKey
+from openai import AzureOpenAI
+from models import AzureOpenAIConfiguration, Message, ModelConfiguration
+
+
+def main():
+    # Create a credential using DefaultAzureCredential
+    credential = DefaultAzureCredential()
+    refresher = None
+
+    # Get the App Configuration endpoint from environment variables
+    app_config_endpoint = os.environ.get("AZURE_APPCONFIGURATION_ENDPOINT")
+    if not app_config_endpoint:
+        raise ValueError(
+            "The environment variable 'AZURE_APPCONFIGURATION_ENDPOINT' is not set or is empty."
+        )
+
+    try:
+        azure_openai_config = None
+        azure_client = None
+
+        def on_refresh_success():
+            azure_openai_config, azure_client = _create_ai_client(provider)
+
+        chat_app_selector = SettingSelector(key_filter="ChatApp:*")
+
+        global provider
+        # Create the configuration provider with refresh settings
+        provider = load(
+            endpoint=app_config_endpoint,
+            selects=[chat_app_selector],
+            credential=credential,
+            keyvault_credential=credential,  # Use the same credential for Key Vault references
+            trim_prefixes=["ChatApp:"],
+            refresh_on=[WatchKey(key="ChatApp:ChatCompletion:Messages")],
+            on_refresh_success=on_refresh_success,
+        )
+
+        azure_openai_config, azure_client = _create_ai_client(provider, credential)
+
+        # Initialize chat conversation
+        chat_conversation = []
+        print("Chat started! What's on your mind?")
+
+        while True:
+            # Refresh the configuration from Azure App Configuration
+            provider.refresh()
+
+            # Configure chat completion with AI configuration
+            chat_completion_config = _extract_chat_completion_config(provider)
+
+            # Get user input
+            user_input = input("You: ")
+
+            # Exit if user input is empty
+            if not user_input.strip():
+                print("Exiting chat. Goodbye!")
+                break
+
+            # Add user message to chat conversation
+            chat_conversation.append({"role": "user", "content": user_input})
+
+            # Get latest system message from AI configuration
+            chat_messages = _get_chat_messages(chat_completion_config)
```

Comment:
I hope this can be simplified to the following if you can get rid of the `Message` type.
```python
chat_messages = list(chat_completion_config.messages)
```

Reply:
We do some validation on the role types in the get chat message section. Are we good with removing it?